src/composables/accountSelector.ts

```diff
@@ -0,0 +1,240 @@
+  const ownAddresses = computed(() => {
+    if (protocolFilter.value) {
+      return accountsGroupedByProtocol.value[protocolFilter.value]?.map((account) => (
+        {
+          name: getName(account.address).value || getDefaultAccountLabel(account),
+          address: account.address,
+          isBookmarked: false,
+          protocol: protocolFilter.value ?? PROTOCOLS.aeternity,
+          type: account.type,
+        }
+      ));
+    }
+    return [];
+  });
+  const accountsFilteredByType = computed(
+    () => {
+      switch (accountSelectType.value) {
+        case ACCOUNT_SELECT_TYPE_FILTER.bookmarked:
+          return addressBookFiltered.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.addressBook:
+          return addressBookFiltered.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.recent:
+          return latestTransactions.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.owned:
+          return ownAddresses.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.all:
+          return [...(addressBookFiltered.value ?? []), ...(ownAddresses.value ?? [])];
+        default:
+          return [];
+      }
+    },
+  );
+  function filterAccountsBookBySearchPhrase(entries: IAccountSelectorEntry[]) {
+    const searchQueryLower = searchQuery.value.toLowerCase();
+    return entries.filter(({ name, address }) => (
+      [name, address].some((val) => val.toLowerCase().includes(searchQueryLower))
+    ));
+  }
+  function sortAccounts(entries: IAccountSelectorEntry[]) {
+    return entries.sort((a, b) => {
+      if (a === b) return 0;
+      return a ? 1 : -1;
+    });
+  }
+  function removeDuplicates(entries: IAccountSelectorEntry[]) {
+    return entries.filter(
+      (addr1, i, addresses) => addresses.findIndex(
+        (addr2) => addr2.address === addr1.address,
+      ) === i,
+    );
+  }
+  const accountsFiltered = computed(
+    () => pipe([
+      filterAccountsBookBySearchPhrase,
+      sortAccounts,
+      removeDuplicates,
+    ])(accountsFilteredByType.value ?? []),
+  );
+
+  function setAccountSelectType(type: AccountSelectTypeFilter, resetProtocolFilter = false) {
+    accountSelectType.value = type;
+    setShowBookmarked(type === ACCOUNT_SELECT_TYPE_FILTER.bookmarked, resetProtocolFilter);
+  }
+  function clearFilters(resetProtocolFilter = false) {
+    accountSelectType.value = ACCOUNT_SELECT_TYPE_FILTER.addressBook;
+    addressBookClearFilters(resetProtocolFilter);
+  }
+
+  watch(
+    () => [rawAddressBookFiltered.value],
+    () => {
+      addressBookFiltered.value = rawAddressBookFiltered.value.map((account) => (
+        {
+          ...account,
+          isAddressBookEntry: true,
+        }
+      ));
+    },
+    { immediate: true }, // Run immediately on initialization
+  );
+  watch(
+    () => [protocolFilter.value],
+    async () => {
+      if (protocolFilter.value) {
+        setAccountSelectType(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+      }
+    },
+  );
+  watch(
```

Comment:
`watch` is actually not light operation, if the code below should be actually calculated once i would rather not use `watch`.

Reply:
The code needs to be re-ran when the active account changes, as well as when there is a new transaction, in order for the `Recent` tab to be updated accordingly

Reply:
From your description @alex-k8 I think the `computed` property is exactly what should be used here. Computed properties knows their dependencies and are updated whenever any of the values changes. The main difference is that it checks the dependencies only if it's needed. And we need those values only when opening the account selector.

The watcher you added will be fired every time user changes the accounts which makes the account switching process more resource consuming.